datafusion/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -414,25 +424,23 @@ impl SortPreservingMergeStream {
                 return Poll::Ready(Err(e));
             }
             Some(Ok(batch)) => {
-                let cursor = Arc::new(
-                    match SortKeyCursor::new(
-                        idx,
-                        self.next_batch_index, // assign this batch an ID
-                        Arc::new(batch),
-                        &self.column_expressions,
-                        self.sort_options.clone(),
-                    ) {
-                        Ok(cursor) => cursor,
-                        Err(e) => {
-                            return Poll::Ready(Err(ArrowError::ExternalError(
-                                Box::new(e),
-                            )));
-                        }
-                    },
-                );
+                let batch = Arc::new(batch);
+                let cursor = match SortKeyCursor::new(
+                    idx,
+                    self.next_batch_index, // assign this batch an ID
```

Comment:
I think now that we have an array called `batches` it will be less confusing for this concept to renamed to `batch_id
`, otherwise it might be confused with an index into `batches` as opposed to an id used to cache comparators.